### PR TITLE
Fix trash from RabbitMQ

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

CMakeLists for AMQP-cpp library turned our build to complete disaster.

CC @alesapin 
CC @azat Look at the warnings from Arrow.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
```
grep -F '"command":' compile_commands.json | grep -v -P ' -c .+?/contrib/' | grep -o -P ' (-Wno-\S+|-w )' | sort | uniq -c | sort -rn | awk '{ if ($1 >= 1000) print $2 }' | sort
```

**Comment by alexey-milovidov** [comment]
Sandbox is totally broken:
```
2020-07-13 04:19:44
Unhandled exception: Traceback (most recent call last):
  File "sandbox/bin/executor.py", line 1043, in _execute
    self.task.on_execute()
  File "sandbox/projects/clickhouse/BaseOnCommitTask/base.py", line 295, in on_execute
    state, description, url = self.process(self.commit, repo, pull_request)
  File "sandbox/projects/clickhouse/BaseOnCommitTask/test_task.py", line 93, in process
    s3_path_prefix + "/" + os.path.basename(raw_log_path))
  File "sandbox/projects/clickhouse/util/s3_helper.py", line 75, in upload_test_report_to_s3
    return self._upload_file_to_s3('clickhouse-test-reports', file_path, s3_path)
  File "sandbox/projects/clickhouse/util/s3_helper.py", line 70, in _upload_file_to_s3
    self.client.upload_file(file_path, bucket_name, s3_path, ExtraArgs=metadata)
  File "contrib/python/boto3/boto3/s3/inject.py", line 131, in upload_file
    extra_args=ExtraArgs, callback=Callback)
  File "contrib/python/boto3/boto3/s3/transfer.py", line 287, in upload_file
    filename, '/'.join([bucket, key]), e))
S3UploadFailedError: Failed to upload ./ClickHouse/utils/check-style/output.txt to clickhouse-test-reports/12448/8f2055b0a019add57278b48cfd7be3384de39d44/style_check/output.txt: An error occurred (ServiceTotalSizeQuotaExceeded) when calling the PutObject operation: You have attempted to exceed the quota for your service on total size of all objects in your buckets. Contact our support to change quotas or remove some objects from your buckets.
```

**Comment by azat** [comment]
>CC @azat Look at the warnings from Arrow.

Ok

File: src/Storages/RabbitMQ/ReadBufferFromRabbitMQConsumer.cpp [hunk 1]
```diff
@@ -146,16 +146,18 @@ void ReadBufferFromRabbitMQConsumer::initExchange()
...
-    binding_arguments["hash-property"] = "message_id";
-
-    /// Declare exchange for sharding.
-    consumer_channel->declareExchange(local_hash_exchange, AMQP::consistent_hash, binding_arguments)
-    .onError([&](const char * message)
     {
-        local_exchange_declared = false;
-        LOG_ERROR(log, "Failed to declare {} exchange: {}", exchange_type, message);
-    });
+        AMQP::Table binding_arguments;
```
Comment (alesapin) [inline src/Storages/RabbitMQ/ReadBufferFromRabbitMQConsumer.cpp:150]: Added scope just to avoid capturing a lot of variables in lambda?

**Review by alesapin — COMMENTED** [review]
Need to fix the release build.

File: src/Storages/RabbitMQ/ReadBufferFromRabbitMQConsumer.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply src/Storages/RabbitMQ/ReadBufferFromRabbitMQConsumer.cpp:150]: -Wshadow

**Comment by azat** [comment]
>Need to fix the release build.

These are -Wdeprecated from arrow, to fix them I need some bits from this PR (`s/PUBLIC/PRIVATE/` + other rabbit mq bits), so I will pick then and fix deprecated API usage for arrow on top of it, and submit it as a separate PR

Or I can just send a patch on top of this PR here, @alexey-milovidov ?

**Comment by alexey-milovidov** [comment]
The most convenient it to create a new branch, then pull all commits from this PR, then add more commits.

**Comment by azat** [comment]
>The most convenient it to create a new branch, then pull all commits from this PR, then add more commits.

@alexey-milovidov #12477

**Pull request merged**